zebra-chain/src/transparent/script.rs

```diff
@@ -12,7 +12,20 @@ use std::{fmt, io};
     any(test, feature = "proptest-impl"),
     derive(proptest_derive::Arbitrary)
 )]
-pub struct Script(pub Vec<u8>);
+pub struct Script(Vec<u8>);
+
+impl Script {
+    /// Create a new Bitcoin script from its raw bytes.
+    /// The raw bytes must not contain the compact size prefix.
+    pub fn new(raw_bytes: Vec<u8>) -> Self {
```

Comment:
Can we be explicit that these are raw bytes, not Zcash serialization?
Also can we restrict these functions to the crate (`pub(crate) fn`), the transparent module (`pub(super) fn`) or the script module (`fn`)?

```suggestion
    fn new_from_raw(raw_bytes: Vec<u8>) -> Self {
```

Reply:
Changed the name and return type.

About the  visibility, good question. It's possible to make it `pub(crate)`, but consider the API: the `Transaction::sighash` method (which is `pub`) receives a `transparent::Output` which in turns needs a `Script`. Therefore, any crate users must be able to construct a `Script` in other to use.

(The reason that it is possible to currently get away with `pub(crate)` is that zebra never seems to call `Transaction::sighash` for transparent inputs, using `zebra_script` instread. But that could change in the future?

Reply:
Ah right, don't worry about the visibility then - it's a type constructor.